Feature/docker publish

danieleguido opened the pull request:
This PR implements Docker build and publish workflow for both the client and server applications. It adds new Dockerfiles for the client and server, a Makefile and a GitHub Actions workflow.

**Dockerization and Build Automation**

* Added `code/Dockerfile.client` and `code/Dockerfile.server`

* Introduced `code/Makefile` with targets to build both client and server images locally, passing in Git and build metadata as arguments for reproducible builds.

**Continuous Integration and Publishing**

* Created `.github/workflows/docker-build-publish-client.yml` to automate building and publishing the **client** Docker image to Docker Hub on pushes to the main branch, ensuring up-to-date images are available.

*  Created `.github/workflows/docker-build-publish-server.yml` to automate building and publishing the **server** Docker image to Docker Hub on pushes to the main branch, ensuring up-to-date images are available.

danieleguido committed:
create gh action

danieleguido committed:
Update .gitignore

danieleguido committed:
Add Dockerfiles and Makefile for client/server builds

Introduces Dockerfile.client and Dockerfile.server for building client and server images with dynamic build metadata. Adds a Makefile to automate Docker builds, passing Git and date information as build arguments for reproducible and informative images.

danieleguido committed:
Update README.md

danieleguido committed:
Update Dockerfile.server

missing tsconfig

danieleguido committed:
fix workflows

sim51 commented:
Hi @danieleguido

Thanks for this PR.

I have a few questions for you. 

Actually the production  is using the docker compose stack, and on each server,  we start a part of the stack. 
If you have the docker images, shall the process will change or not ? 
The compose stack will not be needed anymore, you can just run an image with the good parameters, and it can be easier to maintain, no ?

I don't remember how we have handle the web server that is on the front of the application.
Do we have an nginx (or an other solution) on front of the docker compose stack ?

About your code : 
- we have a docker folder which contains everything that is related to it, and you are not using it.
- the client image you build contains only the build directory (which is great), but there is no web server. So I don't know how you will use it. IMHO, it should contains a web server that serves the website and also do the proxy to the `/graphql` & `/pdf` endpoints.

sim51 committed:
[docker] adding docker for c2dh

- move docker related file inside docker folder
- create a c2dh folder inside the docker folder
- fix server compilation so build folder is autonomus (see
  schema.graphql file and tsconfig)
- adding healtih endpoint on the server

danieleguido committed:
Merge pull request #62 from C2DH/bsi/docker-publish

[docker] adding docker for c2dh

danieleguido committed:
Update Docker setup for environment variable support

Added docker/c2dhunilu/.env to .gitignore and updated docker-compose.yml to use environment variables for sensitive and configurable values. Expanded README with detailed Docker usage instructions, including environment setup and data import steps.
